src/Modules/UI/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js

```diff
@@ -428,16 +428,35 @@ require(['jquery', 'knockout', 'moment', '../util', '../sf', '../config', './../
                     }
                 }
             },
+            /**
+             * in case path is an array, it is expected to be a sorted list of dependent sources.
+             * 
+             * @param path string|array
+             */
             loadBundleScript: function (path) {
-                if (path.indexOf('cdv=') === -1) {
-                    path += (path.indexOf('?') > -1 ? '&' : '?') + 'cdv=' + config.buildNumber;
+
+                function ajax(url, build) {
+                    var isArray = typeof url === 'object' && url.constructor === Array;
+                    $.ajax({
+                        dataType: "script",
+                        cache: true,
+                        data: {
+                            cdv: build
+                        },
+                        url: isArray && url.length > 0 ? url.pop() : url,
+                        complete: function() {
+                            if(isArray && url.length > 0) {
+                                ajax(url, build);
+                            }
+                        }

```

Comment:
Could there be a need for a failure callback?

Reply:
It would be an improvement, since we also have to implement all the code needed to manage that failure. I'd agree if we create a task for it.